cmd/commitment-prefix/main.go

```diff
@@ -3,6 +3,12 @@ package main
 import (
```

Comment:
tree_map = TreeMap()

# Create a new child node
child_node = TreeMapNode()

# Get the parent node
parent_node = tree_map[valsIndex]

# Append the child node to the parent node
parent_node.children.append(child_node)

# Update the TreeMap
tree_map[valsIndex] = parent_node